src/domain/wallets/tx-history.ts

```diff
@@ -65,92 +68,143 @@ export const fromLedger = (
+          feeUsd,
+          type,
+        },
       }
+
+      let txType: ExtendedLedgerTransactionType = type
+      if (type == LedgerTransactionType.IntraLedger && paymentHash) {
+        txType = ExtendedLedgerTransactionType.LnIntraLedger
+      }
+
+      let walletTransaction: WalletTransaction
+      switch (txType) {
+        case ExtendedLedgerTransactionType.IntraLedger:
+          walletTransaction = {
+            ...baseTransaction,
+            initiationVia: {
+              type: PaymentInitiationMethod.IntraLedger,
+              walletId: walletId as WalletId,
+              counterPartyUsername: username as Username,
+            },
+            settlementVia: {
+              type: SettlementMethod.IntraLedger,
+              walletId: walletId as WalletId,
+              counterPartyUsername: username as Username,
+            },
+          }
+          return walletTransaction
+
+        case ExtendedLedgerTransactionType.OnchainIntraLedger:
+          walletTransaction = {
+            ...baseTransaction,
+            initiationVia: {
+              type: PaymentInitiationMethod.OnChain,
+              address,
+            },
+            settlementVia: {
+              type: SettlementMethod.IntraLedger,
+              walletId: walletId as WalletId,
+              counterPartyUsername: username || null,
+            },
+          }
+          return walletTransaction
+
+        case ExtendedLedgerTransactionType.OnchainPayment:
+        case ExtendedLedgerTransactionType.OnchainReceipt:
+          walletTransaction = {
+            ...baseTransaction,
+            initiationVia: {
+              type: PaymentInitiationMethod.OnChain,
+              address,
+            },
+            settlementVia: {
+              type: SettlementMethod.OnChain,
+              transactionHash: txHash as OnChainTxHash,
+            },
+          }
+          return walletTransaction
+
+        case ExtendedLedgerTransactionType.LnIntraLedger:
+          walletTransaction = {
+            ...baseTransaction,
+            initiationVia: {
+              type: PaymentInitiationMethod.Lightning,
+              paymentHash: paymentHash as PaymentHash,
+              pubkey: pubkey as Pubkey,
+            },
+            settlementVia: {
+              type: SettlementMethod.IntraLedger,
+              walletId: walletId as WalletId,
+              counterPartyUsername: username || null,
+            },
+          }
+          return walletTransaction
+
+        case ExtendedLedgerTransactionType.Payment:
+        case ExtendedLedgerTransactionType.Invoice:
+          walletTransaction = {
+            ...baseTransaction,
+            initiationVia: {
+              type: PaymentInitiationMethod.Lightning,
+              paymentHash: paymentHash as PaymentHash,
+              pubkey: pubkey as Pubkey,
+            },
+            settlementVia: {
+              type: SettlementMethod.Lightning,
+              paymentSecret: null,
```

Comment:
should not be `null`?

Reply:
`paymentSecret` should probably be treated differently whether it's a `Payment` of an `Invoice`:

- in the case of an invoice, "we" (ie: our lnd node) actually know the `secret` right from the beginning because we created it. but we should not disclosed it before the invoice is being paid, because we don't want to let a end user know/share it until, this would be an attack vector. ie: an intermediary node could know the secret before it reaches our node, and the payment would never arrived to us so our user would not be credited, but the payer would have sent the money. maybe not a high risk scenario but something to consider.

- in the case of a payment, when we know the secret, it means the invoice has been settled, so there is no risk here

Reply:
True, I can make another type to reflect this (`WalletLnSettledSentTransaction` & `WalletLnSettledReceivedTransaction`) with different names for that property (`paymentSecretGenerated` vs. `paymentSecretReceived` for e.g.)

Reply:
> should not be `null`?

See this [comment](https://github.com/GaloyMoney/galoy/pull/759#discussion_r757837347)